docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+
+**Bootnode** 
+
+A peer that have a public, static ENR to enable new peers to join the network. For the sake of flexibility, 
+bootnode/s ENR values are configurable and can be changed on demand by operators. \
+Bootnode doesn't start a libp2p host for TCP communication, its role ends once a new peer finds existing peers in the network.
+
+#### ENR
+
+[Ethereum Node Records](https://github.com/ethereum/devp2p/blob/master/enr.md) is a format that holds peer information.
+Records contain a signature, sequence (for republishing record) and arbitrary key/value pairs. 
+
+`ENR` structure in `SSV.Network` consists of the following key/value pairs:
+
+| Key         | Value                                                          | Status          |
+|:------------|:---------------------------------------------------------------|:----------------|
+| `id`        | name of identity scheme, e.g. "v4"                             | Done            |
+| `secp256k1` | compressed secp256k1 public key, 33 bytes                      | Done            |
+| `ip`        | IPv4 address, 4 bytes                                          | Done            |
+| `tcp`       | TCP port, big endian integer                                   | Done            |
+| `udp`       | UDP port, big endian integer                                   | Done            |
+| `type`      | node type, integer; 1 (operator), 2 (exporter), 3 (bootnode)   | Done (`v0.1.9`) |
+| `oid`       | operator id, 32 bytes                                          | Done (`v0.1.9`) |
+| `version`   | fork version, integer                                          | -               |
+| `subnets`   | bitlist, 0 for irrelevant and 1 for assigned subnet            | -               |
+
+#### Discovery Alternatives
+
+[libp2p's Kademlia DHT](https://github.com/libp2p/specs/tree/master/kad-dht) offers similar features, and even a more complete implemetation of Kademlia DHT.
+Discv5 design is loosely inspired by the Kademlia DHT, but unlike most DHTs no arbitrary keys and values are stored. Instead, the DHT stores and relays node records.
+
+Libp2p's Kad DHT will allow advertising and finding peers by multiple keys, e.g. topics/subnets.
+As `ENR` has a size limit (`< 300` bytes), and therefore discv5 won't support multiple key/value pairs.
+
+Notes:
+- discv5 specifies [Topic Index](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-rationale.md#the-topic-index) that help to lookup relevant nodes in a smaller set of nodes, currently not fully implemented
+- See [Ethereum specs > phase 0 > p2p interface](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-discv5-and-not-libp2p-kademlia-dht) 
+details why discv5 was chosen over libp2p Kad DHT in Ethereum.
+
+In `v0` discv5 is used, `v1` TBD, this section will be updated once that work is complete.
+
+### Network ID
```

Comment:
There are 2 use cases for network ID.
1) in networking which is just to not spam and connect with peers from the wrong network
2) in signing messages so that if we by accident have the same private key for testnet and mainnet we won't have valid testnet sigs on mainnet.

Where are we using the network ID?